tests/test_buku.py

```diff
@@ -997,3 +1003,28 @@ def test_parse_range(tokens, valid, expected):
         except Exception as e:
             assert type(e) is type(expected)
             assert str(e) == str(expected)
+
+
+def test_split_by_marker():
+    search_string = (' global substring  .title substring :url substring :https '
+                     '> description substring #partial,tags: #,exact,tags, *another global substring ')
+    assert split_by_marker(search_string) == [
+        ' global substring', '.title substring', ':url substring', ':https',
+        '> description substring', '#partial,tags:', '#,exact,tags,', '*another global substring ',
+    ]
```

Comment:
The split condition is “one or multiple spaces followed by a marker” (with the marker being retained)